ci.sh

```diff
@@ -37,25 +39,53 @@ python -c "import sys, struct, ssl; print('python:', sys.version); print('versio
 echo "::endgroup::"
 
 echo "::group::Install dependencies"
-python -m pip install -U pip build
+python -m pip install -U pip uv
 python -m pip --version
+uv --version
+
+# expands to 0 != 1 if MAKE_VENV is not set, if set the `-0` has no effect
+# https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_02
+if [ "${MAKE_VENV-0}" == 1 ]; then
+    uv venv .venv --seed
+
+    if [ "$ON_WINDOWS" = "True" ]; then
+        source ./.venv/Scripts/activate
+    else
+        source .venv/bin/activate
+    fi
```

Comment:
I'm skeptical we need to keep the `MAKE_VENV` code at all, if we want isolated build&testing environment locally we should be using tox (though uv plans to reimplement that themselves at some point :tm:).
But I guess I'm fine with leaving this around until then, even though I see very marginal use for it.

Ideally environmental variables like this should be documented in the contributing guide, if we expect developers to use it at all. If this is only a CI thing... then we truly can remove it since you don't seem to set it in the CI config?

The contributor docs should maybe also be changed to instruct users to use `uv` rather than pip directly, don't think it makes a difference now but might break in the future if we start using uv-specific features?

Reply:
The reason I left it is because on my machine, I installed python with system package manager, which causes both pip and uv to raise warnings about how modifying system python installation can be harmful, and using a virtual environment means that doesn't happen.

Reply:
whenever working with packages it's highly recommended you as a developer create a separate virtualenv for each repo. I try to have that as a step in any contributor docs I write/edit, but maybe haven't given trio that treatment

Reply:
https://trio.readthedocs.io/en/stable/contributing.html#tests does only a weak suggestion atm... that's easy to miss due to the length of the line 🙃:

̀̀```
cd path/to/trio/checkout/
pip install -r test-requirements.txt # possibly using a virtualenv
pytest trio 
```

I'd change that to

̀̀```
cd path/to/trio/checkout/
python -m venv .venv
source .venv/bin/activate
pip install -r test-requirements.txt
pytest trio 
```

Reply:
Removed `MAKE_ENV` in 57b8d2f